src/app/_components/Event/EventFilter.js

```diff
@@ -0,0 +1,179 @@
+    getNextPageParam: lastPage => lastPage?.metaData.lastCursor,
+  });
+
+  // Get data on first render or on url open
+  useEffect(() => {
+    const monthFromQuery = searchParams.get('month');
+
+    if (monthFromQuery && monthFromQuery !== currentMonth) {
+      // If the month is defined in the query and is not equal to the current month
+      setActiveMonthNumber(monthFromQuery);
+      setActiveIndex(monthFromQuery - 1);
+      setActiveMonth(months.filter(item => item.index === parseInt(monthFromQuery, 10)));
+      allEvents({ month: monthFromQuery, take: '', lastCursor: '' });
+    } else if (!monthFromQuery || monthFromQuery === undefined) {
+      setActiveMonth(months[0]);
+      setActiveMonthNumber(currentMonth);
+      setActiveIndex(currentMonth - 1);
+      deleteParam();
+      addParam(currentMonth.toString());
+      allEvents({ month: currentMonth, take: '', lastCursor: '' });
+    }
+
+    // eslint-disable-next-line
+  }, [currentMonth]);
+
+  useEffect(() => {
+    // if the last element is in view and there is a next page, fetch the next page
+    if (inView && hasNextPage) {
+      fetchNextPage();
+    }
+    // eslint-disable-next-line
+  }, [hasNextPage, inView]);
+
+  // // Filter data based on selected month
+  const handleFilter = newMonth => {
+    setActiveIndex(newMonth.index);
+    setActiveMonth(newMonth);
+    const calcMonth = parseInt(newMonth.index + 1, 10);
+    setActiveMonthNumber(calcMonth);
+    deleteParam();
+    addParam(calcMonth);
+    allEvents({ month: calcMonth, take: '', lastCursor: '' });
+  };
+
+  const handleMouseEnter = index => {
+    const newHoveredIndexes = [...hoveredIndexes];
+    newHoveredIndexes[index] = true;
+    setHoveredIndexes(newHoveredIndexes);
+  };
+
+  const handleMouseLeave = index => {
+    const newHoveredIndexes = [...hoveredIndexes];
+    newHoveredIndexes[index] = false;
+    setHoveredIndexes(newHoveredIndexes);
+  };
+
+  if (error) return <div className="mt-10">{('An error has occurred: ', error.message)}</div>;
+
+  return (
+    <>
+      <div className="lg:w-max-[900px] mx-auto flex w-full flex-col items-start justify-start gap-6 self-stretch">
+        <div className="flex flex-row flex-wrap items-start justify-start gap-3">
+          {months.map(month => (
+            <PillButton
+              variant="eventFilter"
+              colorVariant="semiorange"
+              className={cn(activeIndex === month.index && activeButtonStyles, 'group w-fit')}
+              key={month.index}
+              activemonth={activeMonth}
+              onMouseEnter={() => handleMouseEnter(month.index)}
+              onMouseLeave={() => handleMouseLeave(month.index)}
+              onFocus={() => {
+                handleMouseEnter(month.index);
+              }}
+              onBlur={() => handleMouseLeave(month.index)}
+              onMouseDown={() => handleMouseEnter(month.index)}
+              onMouseUp={() => handleMouseLeave(month.index)}
```

Comment:
@Yivgha we should avoid all of this event handlers using native hover

Reply:
changed